### PR TITLE
WIP 71 rss feed

### PR DESCRIPTION
**Pull request by jum-s**

### CODE REVIEW

**Comment by jum-s** [comment]
TODO

- [x]  Query on all public entities what so ever
- [ ] Query on user's network private entities (see [Token Private Feeds](https://github.com/blog/16-token-private-feeds)

File: npm-debug.log [hunk 1]
```diff
@@ -0,0 +1,47 @@
+0 info it worked if it ends with ok
```
Comment (maxlath) [inline npm-debug.log]: this file should be `git rm`

File: server/controllers/feeds/feeds.coffee [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    { query } = req
+    items_.publicByDate limit
+    .then (items) -> rssSerializer(feed, items)
+    .then -> res.send(feed.xml())
+    .catch error_.Handler(req, res)
+
+rssSerializer = (feed, items)->
+  items.map (item)->
+    title: item["title"]
+    guid: item["entity"]
```
Comment (maxlath) [inline server/controllers/feeds/feeds.coffee]: the item guid should be `item._id` (which is a CouchDB uuid), not it's `item.entity` which will be the same on all items from the same entity

File: server/controllers/feeds/feeds.coffee [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+_ = __.require 'builders', 'utils'
+root = CONFIG.fullPublicHost()
+items_ = __.require 'controllers', 'items/lib/items'
+error_ = __.require 'lib', 'error/error'
+RSS = require 'rss'
+limit = 100
+
+module.exports =
+  get: (req, res, next)->
+    feed = new RSS({"title":"New Public Items"})
```
Comment (maxlath) [inline server/controllers/feeds/feeds.coffee]: prefered style:
```coffee
feed = new Rss { title: 'New Public Items' }
```

File: server/controllers/feeds/feeds.coffee [hunk 4]
```diff
@@ -0,0 +1,27 @@
+CONFIG = require 'config'
+__ = require('config').universalPath
+_ = __.require 'builders', 'utils'
+root = CONFIG.fullPublicHost()
+items_ = __.require 'controllers', 'items/lib/items'
+error_ = __.require 'lib', 'error/error'
+RSS = require 'rss'
```
Comment (maxlath) [inline server/controllers/feeds/feeds.coffee]: prefered style:
```coffee
Rss = require 'rss'
```
The capital indicates a constructor

File: server/controllers/feeds/feeds.coffee [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    feed = new RSS({"title":"New Public Items"})
+    { query } = req
+    items_.publicByDate limit
+    .then (items) -> rssSerializer(feed, items)
+    .then -> res.send(feed.xml())
+    .catch error_.Handler(req, res)
+
+rssSerializer = (feed, items)->
+  items.map (item)->
+    title: item["title"]
```
Comment (maxlath) [inline server/controllers/feeds/feeds.coffee]: prefered style:
```coffee
title: item.title
```

File: server/controllers/feeds/feeds.coffee [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+items_ = __.require 'controllers', 'items/lib/items'
+error_ = __.require 'lib', 'error/error'
+RSS = require 'rss'
+limit = 100
+
+module.exports =
+  get: (req, res, next)->
+    feed = new RSS({"title":"New Public Items"})
+    { query } = req
+    items_.publicByDate limit
```
Comment (maxlath) [inline server/controllers/feeds/feeds.coffee]: better start with user-specific public inventories than with the global public inventory, which sole interest is to display examples on [/welcome](http://inventaire.io/welcome)

Comment (jum-s) [inline server/controllers/feeds/feeds.coffee]: `publicListings` ?

Comment (maxlath) [inline server/controllers/feeds/feeds.coffee]: yep, [publicListings](https://github.com/inventaire/inventaire/blob/6fa25ab1c80b3449edd856b7dbe8cd1ce3365dd4/server/controllers/items/lib/items.coffee#L41) seems the right function:
```coffee
{ user:userId } = query
# items_.publicListings expects an array of user ids
items_.publicListings [ userId ]
```

**Comment by maxlath** [comment]
@jum-s I enriched the feeds and items data, review welcome!

**Comment by maxlath** [comment]
remaining TODOs:
- [ ] user-specific aggregated friends and groups feed
- [ ] nearby public items feed

**Pull request merged**